app/forms/floating-ip-edit.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useNavigate, type LoaderFunctionArgs } from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQueryClient,
+  usePrefetchedApiQuery,
+} from '@oxide/api'
+import {
+  Listbox,
+  Networking16Icon,
+  PropertiesTable,
+  ResourceLabel,
+  Truncate,
+} from '@oxide/ui'
+import { formatDateTime } from '@oxide/util'
+
+import { SideModalForm } from 'app/components/form'
+import { getFloatingIpSelector, useFloatingIpSelector, useForm } from 'app/hooks'
+import { addToast } from 'app/stores/toast'
+import { pb } from 'app/util/path-builder'
+
+EditFloatingIpSideModalForm.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project, floatingIp } = getFloatingIpSelector(params)
+  await Promise.all([
+    apiQueryClient.prefetchQuery('floatingIpView', {
+      path: { floatingIp },
+      query: { project },
+    }),
+    apiQueryClient.prefetchQuery('instanceList', {
+      query: { project },
+    }),
+    apiQueryClient.prefetchQuery('floatingIpList', {
+      query: { project },
+    }),
+  ])
+  return null
+}
+
+export function EditFloatingIpSideModalForm() {
+  const { project, floatingIp: floatingIpName } = useFloatingIpSelector()
+  const { data: floatingIp } = usePrefetchedApiQuery('floatingIpView', {
+    path: { floatingIp: floatingIpName },
+    query: { project },
+  })
+  const { data: instances } = usePrefetchedApiQuery('instanceList', {
+    query: { project },
+  })
+  const { data: floatingIps } = usePrefetchedApiQuery('floatingIpList', {
+    query: { project },
+  })
+  console.log(instances, floatingIps)
+
+  const dismissLink = pb.floatingIps({ project })
+  const navigate = useNavigate()
+  const form = useForm({ defaultValues: { instanceId: floatingIp.instanceId } })
```

Comment:
Changing this to `instanceId: floatingIp.instanceId || null` will fix the type error on the `selected` prop below by making this `string | null` instead of `string | undefined`.

In other forms we tend to have an explicit type for the form values. I think that makes it easier to think about the form. Often we can use the `*Create` type from the API directly or modify it slightly, but here we can't, so I would just make it from scratch:

```ts
type FormValues = {
  instanceId: string | null
}
```

And then on this line you can do something like

```ts
const form = useForm({ 
  defaultValues: { instanceId: floatingIp.instanceId || null } satisfies FormValues
})
```